src/devices/OpenHardwareMonitor/OpenHardwareMonitor.cs

```diff
@@ -0,0 +1,740 @@
+    /// </summary>
+    public sealed class OpenHardwareMonitor : IDisposable
+    {
+        private static readonly TimeSpan MonitorInterval = TimeSpan.FromMilliseconds(100);
+
+        private delegate IQuantity UnitCreator(float value);
+
+        /// <summary>
+        /// Event that gets invoked when a value is updated
+        /// </summary>
+        /// <param name="sensor">Sensor that has an updated value</param>
+        /// <param name="value">New value for the sensor</param>
+        /// <param name="timeSinceUpdate">Time since the last update of this sensor</param>
+        public delegate void OnNewValue(Sensor sensor, IQuantity value, TimeSpan timeSinceUpdate);
+
+        private static Dictionary<SensorType, (Type Type, UnitCreator Creator)> _typeMap;
+        private Hardware _cpu;
+        private Hardware _gpu;
+
+        private Thread _monitorThread;
+        private object _lock;
+        private List<MonitoringJob> _monitoredElements;
+        private List<Sensor> _derivedSensors;
+        private DateTimeOffset _lastMonitorLoop;
+
+        static OpenHardwareMonitor()
+        {
+            _typeMap = new Dictionary<SensorType, (Type Type, UnitCreator Creator)>();
+            _typeMap.Add(SensorType.Temperature, (typeof(Temperature), (x) => Temperature.FromDegreesCelsius(x)));
+            _typeMap.Add(SensorType.Voltage, (typeof(ElectricPotential), x => ElectricPotential.FromVolts(x)));
+            _typeMap.Add(SensorType.Load, (typeof(Ratio), x => Ratio.FromPercent(x)));
+            _typeMap.Add(SensorType.Fan, (typeof(RotationalSpeed), x => RotationalSpeed.FromRevolutionsPerMinute(x)));
+            _typeMap.Add(SensorType.Flow, (typeof(VolumeFlow), x => VolumeFlow.FromLitersPerHour(x)));
+            _typeMap.Add(SensorType.Control, (typeof(Ratio), x => Ratio.FromPercent(x)));
+            _typeMap.Add(SensorType.Level, (typeof(Ratio), x => Ratio.FromPercent(x)));
+            _typeMap.Add(SensorType.Power, (typeof(Power), x => Power.FromWatts(x)));
+            _typeMap.Add(SensorType.Clock, (typeof(Frequency), x => Frequency.FromMegahertz(x)));
+            _typeMap.Add(SensorType.Energy, (typeof(Energy), x => Energy.FromWattHours(x)));
+            _typeMap.Add(SensorType.HeatFlux, (typeof(HeatFlux), x => HeatFlux.FromKilowattsPerSquareMeter(x)));
+            _typeMap.Add(SensorType.Current, (typeof(ElectricCurrent), x => ElectricCurrent.FromAmperes(x)));
+        }
+
+        /// <summary>
+        /// Constructs a new instance of this class.
+        /// The class can be constructed even if no sensors are available or OpenHardwareMonitor is not running (yet).
+        /// </summary>
+        /// <exception cref="PlatformNotSupportedException">The operating system is not Windows.</exception>
+        public OpenHardwareMonitor()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I will check what it takes to use it on linux (and at least document that).

Reply:
There seems to be some sort of support on linux, using mono. I haven't tried that, though. There's an alternate connection api that creates a small HTTP server that could be used, but this would need to be investigated separately.